apps/spruce/src/pages/image/PackagesTable/index.tsx

```diff
@@ -0,0 +1,138 @@
+import { useMemo, useRef, useState } from "react";
+import { useQuery } from "@apollo/client";
+import styled from "@emotion/styled";
+import Pagination from "@leafygreen-ui/pagination";
+import {
+  useLeafyGreenTable,
+  LGColumnDef,
+  ColumnFiltersState,
+} from "@leafygreen-ui/table";
+import { BaseTable } from "components/Table/BaseTable";
+import { size } from "constants/tokens";
+import { useToastContext } from "context/toast";
+import {
+  Package,
+  ImagePackagesQuery,
+  ImagePackagesQueryVariables,
+} from "gql/generated/types";
+import { IMAGE_PACKAGES } from "gql/queries";
+
+type PackagesTableProps = {
+  imageId: string;
+};
+
+const PAGE_SIZE = 10;
+
+export const PackagesTable: React.FC<PackagesTableProps> = ({ imageId }) => {
+  const dispatchToast = useToastContext();
+  const [pagination, setPagination] = useState({
+    pageIndex: 0,
+    pageSize: PAGE_SIZE,
+  });
+  const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([]);
+  const { data: packagesData, loading } = useQuery<
+    ImagePackagesQuery,
+    ImagePackagesQueryVariables
+  >(IMAGE_PACKAGES, {
+    variables: {
+      imageId,
+      opts: {
+        page: pagination.pageIndex,
+        limit: pagination.pageSize,
+        name: columnFilters.find((filter) => filter.id === "name")
+          ?.value as string,
+      },
+    },
+    onError(err) {
+      dispatchToast.error(
+        `There was an error loading image packages: ${err.message}`,
+      );
+    },
+  });
+
+  const packageEntries = useMemo(
+    () => packagesData?.image?.packages.data ?? [],
+    [packagesData?.image?.packages.data],
+  );
+
+  const totalNumPackages = useMemo(
+    () => packagesData?.image?.packages.totalCount ?? 0,
+    [packagesData?.image?.packages.totalCount],
+  );
```

Comment:
The count doesn't appear to be correct when a filter is applied:

<img width="725" alt="Screenshot 2024-08-15 at 7 36 07 PM" src="https://github.com/user-attachments/assets/33220ad6-5b11-4fae-a07c-c84e58cf1410">

(Specifically there shouldn't be 103 pages, and we shouldn't be able to click the Next button)